dotCMS/src/main/java/com/dotmarketing/portlets/structure/model/ContentletRelationships.java

```diff
@@ -134,9 +135,30 @@ private boolean isShowUpField(ContentletRelationshipRecords records, String fiel
 	public List<ContentletRelationshipRecords> getLegacyRelationshipsRecords() {
 
 		//Filter legacy relationship records, which do not have a relationship field
-		return relationshipsRecords.stream()
+		final List<ContentletRelationshipRecords> recordsToDisplay = relationshipsRecords.stream()
 				.filter(record -> !record.getRelationship().isRelationshipField())
 				.collect(CollectionsUtils.toImmutableList());
+
+		//if legacy relationships are found, the whole list of relationships without fields is populated
+		if (UtilMethods.isSet(recordsToDisplay)){
+			final ContentType contentType = contentlet.getContentType();
+
+			//Obtain field relationships
+			final List<ContentletRelationshipRecords> fieldRelationships = contentType.fields().stream()

```

Comment:
looks like this should be a Set in order to get a faster contains

Reply:
I would have to implement a comparator because the order needs to be guaranteed